### DISCUSSION TITLE
alevin: is it possible to change order of UMI and CB in R1?

### DISCUSSION THREAD

**Discussion by DrHogart** [COMBINE-lab/salmon#493] [Q&A (single-cell specific)]
Hi,
I'm using the custom 3'-tag data protocol and in my scheme R1 contains barcodes in the following order: 5'-UMI-CB. But as far as I understand, the alevin can deal only with this order: 5'-CB-UMI. Is it possible to add the feature to alevin allowing to customize the order of barcodes in the R1?

**Comment by k3yavi** [comment]
Thanks @DrHogart for your interest in alevin.
I think it will be useful feature to have define the order of UMI & CB, we will have that, most probably, in the next version of salmon. Hopefully it will be released by coming week, along with other features.

The idea would be to use `--end` flag. Currently it has been set to 5 reflecting 5'.
We already have CelSeq (6 CB,6 UMI) and CelSeq2 (8CB, 6UMI) single cell protocol flags in the current build of alevin, which assumes UMI first however if your custom protocol uses longer length CB/UMI, you can use alevin with following command with the next salmon release :

`--barcodeLength X --umiLength Y --end 3` where X being CB and Y being UMI lengths.

**Comment by DrHogart** [comment]
Thanks for the suggestion. However, besides the CB and UMI I have some other nucleotides in the R1 (5'-UMI-CB-Xn-3'), so `--end 3` will not work for me. Currently, I just swap the CB and UMI in my R1, so alevin can parse them. Looking forward to see the update of salmon. Thanks for your work!
